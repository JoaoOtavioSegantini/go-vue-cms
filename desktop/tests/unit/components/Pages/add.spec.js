import { mount } from "@vue/test-utils";
import AddPage from "@/components/Pages/Add.vue";
import ClassicEditor from "@ckeditor/ckeditor5-build-classic";
import { createStore } from "vuex";

 window.axios = {
    post: jest
      .fn(()  => {
        return {
          then: jest.fn(() => "your faked response"),
        };
      })
      .mockName("axiosPost"),
  };

describe("AddPage.vue", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  const mockRouter = {
    push: jest.fn(),
  };
  const createVuexStore = () => {
    return createStore({
      state() {
        return {
          all: [{ ID: "1", slug: "John Doe", title: "johndoe@gmail.com" }],
        };
      },
      mutations: {
        addToPagesList(state, res) {
          state.all.push(res);
        },
      },
      actions: {
        createPage(ctx, res) {
          ctx.commit("addToPagesList", res);
        },
      },
    });
  };

  function factory() {
    const store = createVuexStore();

    return mount(AddPage, {
        stubs: {
            ckeditor: true
          },
      
      mocks: {
        axios: window.axios,
      },
      global: {
        plugins: [store],
        mocks: {
          $router: mockRouter,
        //   $route: mockRoute,
        },
      },
      data() {
        return {
          editor: ClassicEditor,
          page: { ID: "1", title: "page 1" },
          editorData: "<p>Content of the editor.</p>",
        };
      },
    });
  }

  it("renders props.msg when passed", async () => {
    const msg =
      "AÇÕES Nova página Inclusão de nova página no siteTítuloURLConteúdoSalvar";
    const wrapper = factory();

    wrapper.findAll("input")[0].setValue("acbdfg@gmail.com");
    wrapper.findAll("input")[1].setValue("987654");

    await wrapper.find("form").trigger("submit");
    await wrapper.find("button").trigger("click");

    await wrapper.vm.save()
    console.log(wrapper.emitted())

    //expect(window.axios.post).toBeCalledWith({});
    expect(mockRouter.push).toBeCalledWith({ "path": "/pages"});
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
