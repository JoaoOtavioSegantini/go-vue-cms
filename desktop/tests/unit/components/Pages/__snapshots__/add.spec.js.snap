// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPage.vue renders props.msg when passed 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col-1 nav\\">
      <p class=\\"display-6\\"><small>AÇÕES</small></p><a href=\\"/pages\\" class=\\"btn btn-secondary\\" title=\\"Voltar\\"><i class=\\"fa fa-arrow-circle-left\\"></i></a>
    </div>
    <div class=\\"col-11\\">
      <div class=\\"card\\">
        <div class=\\"card-body\\">
          <h5 class=\\"title\\"><i class=\\"fa fa-file-text-o\\"></i> Nova página <small>Inclusão de nova página no site</small></h5>
          <form>
            <div class=\\"form-group\\"><label for=\\"title\\">Título</label><input type=\\"text\\" class=\\"form-control\\" id=\\"title\\" placeholder=\\"Título da página\\"></div>
            <div class=\\"form-group\\"><label for=\\"slug\\">URL</label><input type=\\"text\\" class=\\"form-control\\" id=\\"slug\\" placeholder=\\"Url da página\\"></div>
            <div class=\\"form-group\\"><label for=\\"body\\">Conteúdo</label>
              <ckeditor editor=\\"class nC extends Ng{}\\" modelvalue=\\"<p>Content of the editor.</p>\\"></ckeditor>
            </div><button type=\\"submit\\" class=\\"btn btn-primary\\">Salvar</button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
